src/uct/cuda/gdr_copy/gdr_copy_md.c

```diff
@@ -30,77 +54,399 @@ static ucs_status_t uct_gdr_copy_md_query(uct_md_h md, uct_md_attr_t *md_attr)
 static ucs_status_t uct_gdr_copy_mkey_pack(uct_md_h md, uct_mem_h memh,
                                            void *rkey_buffer)
 {
+    uct_gdr_copy_key_t *packed      = (uct_gdr_copy_key_t *)rkey_buffer;
+    uct_gdr_copy_mem_t *mem_hndl    = (uct_gdr_copy_mem_t *)memh;
+
+    packed->vaddr   = mem_hndl->info.va;
+    packed->bar_ptr = mem_hndl->bar_ptr;
+
     return UCS_OK;
 }
 
 static ucs_status_t uct_gdr_copy_rkey_unpack(uct_md_component_t *mdc,
                                              const void *rkey_buffer, uct_rkey_t *rkey_p,
                                              void **handle_p)
 {
-    *rkey_p   = 0xdeadbeef;
+    uct_gdr_copy_key_t *packed = (uct_gdr_copy_key_t *)rkey_buffer;
+    uct_gdr_copy_key_t *key;
+
+    key = ucs_malloc(sizeof(uct_gdr_copy_key_t), "uct_gdr_copy_key_t");
+    if (NULL == key) {
+        ucs_error("Failed to allocate memory for uct_gdr_copy_key_t");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    key->vaddr      = packed->vaddr;
+    key->bar_ptr    = packed->bar_ptr;
+
     *handle_p = NULL;
+    *rkey_p   = (uintptr_t)key;
+
     return UCS_OK;
 }
 
 static ucs_status_t uct_gdr_copy_rkey_release(uct_md_component_t *mdc, uct_rkey_t rkey,
                                               void *handle)
 {
+    ucs_assert(NULL == handle);
+    ucs_free((void *)rkey);
```

Comment:
no need to cast

Reply:
not fixed

Reply:
its  compiler warning without cast. this is inline with other MDs rkey free